integration-tests/test/console.js

```diff
@@ -1,3 +1,47 @@
+import { scanner } from "./lib/sources.js";
+import { CONSOLE } from "./lib/MiniAppType.js";
+import {
+  INITIAL_STATUS_MESSAGES,
+  EXIT_STATUS_MESSAGE,
+  assertMessagesEqual,
+  verifyMessages
+} from "./helpers/testHelpers.js";
+
 describe("Console", () => {
-  // TODO
+  it("Runs simple interactive console project", (done) => {
+    const expectedMessages = [
+      ...INITIAL_STATUS_MESSAGES,
+      {type: "SYSTEM_OUT", value: "What's your name?"},
+      {type: "SYSTEM_OUT", value: "\n"},
+      {messageType: "SYSTEM_IN", message: "Ben"},
+      {type: "SYSTEM_OUT", value: "Hello Ben!"},
+      {type: "SYSTEM_OUT", value: "\n"},
+      EXIT_STATUS_MESSAGE
+    ];
+    const assertOnMessagesObserved = observedMessages => assertMessagesEqual(observedMessages, expectedMessages);
+
+    let hasReceivedMessage, hasReceivedNewLineAfterMessage;
+    const onMessageCallback = (parsedData, socket, allMessages) => {
+      // Confirm receipt of "What's your name" and new line messages back to back.
+      if (parsedData.type === "SYSTEM_OUT" && parsedData.value === "What's your name?") {
```

Comment:
should we have constants for `"What's your name"` and `"Ben"`?

Reply:
This whole implementation is tightly coupled to the sources code provided -- I thought a little about whether we should modify sources.js to include expected messages like this one, but ended up deciding it didn't feel worth doing at the moment?

Reply:
Yeah I think it's fair to make assumptions given the source code we're using - we do that for example in the Hello World test where we expect a specific string ("Hello World")